File: a-common-sense-guide-of-data-structures-and-algorithms/chapter-8-exercises/exercise-4.py
def nonDuplicated(string):
  hashTable = {}

  for value in string:
    if hashTable.get(value):
      hashTable[value] = hashTable[value] + 1
    else:
      hashTable[value] = 1
    

  for value in string:
    if hashTable[value] == 1:
      return value

  return None


print(nonDuplicated("minimum"))

File: a-common-sense-guide-of-data-structures-and-algorithms/chapter-8-exercises/exercise-4.js
function nonDuplicated(string) {
  let hashTable = {}

  for (const value of string) {
    if (hashTable[value]) {
      hashTable[value] = hashTable[value] + 1
    } else {
      hashTable[value] = 1
    }
  }

  for (const value of string) {
    if (hashTable[value] === 1) {
      return value
    }
  }

  return null
}

console.log(nonDuplicated("minimum"))